### PR TITLE
optimize local development install script

### PR DESCRIPTION
**Pull request by mikechengwei**
Signed-off-by: mikechengwei <842725815@qq.com>

### What problem does this PR solve?
#1027
When local development , we need to execute multiple steps install chaos mesh.
I think we can provide a  script that only install once.

### What is changed and how does it work?
I merge below steps into `local-up-operator.sh` script.

1. Build the image
2. Load image into Kubernetes nodes
3. Deploy Chaos Mesh

User only need to execute `local-up-operator.sh` script when reinstall chaos-mesh.

### Checklist

Tests

- [x] Manual test (add detailed scripts or steps below)

### CODE REVIEW

**Comment by zhouqiang-cl** [comment]
@cwen0 @WangXiangUSTC PTAL

File: hack/local-up-operator.sh [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+      installOnly=true
+        ;;
+    esac
+done
+
+PROVIDER=${PROVIDER:-kind}
+CLUSTER=${CLUSTER:-}
+KUBECONFIG=${KUBECONFIG:-~/.kube/config}
+KUBECONTEXT=${KUBECONTEXT:-}
+DOCKER_REGISTRY_PREFIX=${DOCKER_REGISTRY_PREFIX:-localhost:5000}
```
Comment (fewdan) [inline hack/local-up-operator.sh:66]: Is it necessary to set the default value to "localhost:5000" here?

Comment (mikechengwei) [reply hack/local-up-operator.sh:66]: I think local registry always is localhost:5000, so we can set it as default value.

File: install.sh [hunk 2]
```diff
@@ -1101,7 +1107,7 @@ spec:
       hostPID: true
       containers:
         - name: chaos-daemon
-          image: pingcap/chaos-daemon:${VERSION_TAG}
+          image: ${DOCKER_REGISTRY_PREFIX}/pingcap/chaos-daemon:${VERSION_TAG}
```
Comment (WangXiangUSTC) [inline install.sh:1110]: when `DOCKER_REGISTRY_PREFIX` is empty, `/pingcap/chaos-daemon:latest` will be an invalid image name.****

Comment (mikechengwei) [reply install.sh:1110]: Ok , i will optimize it.

Comment (mikechengwei) [reply install.sh:1110]: I have set default value `docker.io`

**Comment by mikechengwei** [comment]
@cwen0 @WangXiangUSTC  PLTA

**Review by WangXiangUSTC — COMMENTED** [review]
rest LGTM

File: hack/local-up-operator.sh [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+This commands run chaos-mesh in Kubernetes.
+
+Usage: hack/local-up-operator.sh [-hd]
+
+    -h      show this message and exit
+    -i      install dependencies only
+
+Environments:
+
+    PROVIDER              Kubernetes provider. Defaults: kind.
```
[the remark this answers is not in the document]
Comment (mikechengwei) [reply hack/local-up-operator.sh:36]: Now, we only support kind.

[the remark this answers is not in the document]
Comment (mikechengwei) [reply hack/local-up-operator.sh:36]: So, there is no need to add other providers.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -121,28 +121,10 @@ $ make test
...
-   $ kind load docker-image pingcap/chaos-mesh:latest
-   $ kind load docker-image pingcap/chaos-dashboard:latest
-   $ kind load docker-image pingcap/chaos-daemon:latest
-   ```
-
-5. Deploy Chaos Mesh:
-
-   ```bash
-   $ ./install.sh --runtime containerd
+   $ ./hack/local-up-operator.sh
```
[the remark this answers is not in the document]
Comment (mikechengwei) [reply CONTRIBUTING.md:130]: ok

File: hack/local-up-operator.sh [hunk 5]
```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env bash
```
Comment (cwen0) [inline hack/local-up-operator.sh:1]: This script is very useful for developers! Thanks!

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -121,28 +121,13 @@ $ make test
...
-   $ kind load docker-image pingcap/chaos-daemon:latest
-   ```
-
-5. Deploy Chaos Mesh:
+   Following command will rebuild project code and reinstall chaos mesh.
 
    ```bash
-   $ ./install.sh --runtime containerd
+   #
+   $ ./hack/local-up-operator.sh
```
Comment (cwen0) [inline CONTRIBUTING.md:130]: How about renaming this file to `local-up-chaos-mesh.sh`?

**Comment by WangXiangUSTC** [comment]
LGTM

File: install.sh [hunk 7]
```diff
@@ -610,13 +614,13 @@ install_chaos_mesh() {
     local runtime=$7
     local k3s=$8
     local version=$9
-    local timezone=$10
-
+    local timezone=${10}
+    local docker_registry=${11}
```
Comment (cwen0) [inline install.sh:618]:
```
    local timezone=$10

    local docker_registry=$11
```
Keep a uniform format

File: install.sh [hunk 8]
```diff
@@ -630,8 +634,8 @@ install_chaos_mesh() {
     fi
 
     gen_crd_manifests "${crd}" | kubectl apply -f - || exit 1
-    gen_chaos_mesh_manifests "${runtime}" "${k3s}" "${version}" "${timezone}" "${host_network}" | kubectl apply -f - || exit 1
-}
+    gen_chaos_mesh_manifests "${runtime}" "${k3s}" "${version}" "${timezone}" "${host_network}" "${docker_registry}" | kubectl apply -f - || exit 1
+  }
```
Comment (cwen0) [inline install.sh:638]: Remove this blank here.

File: hack/local-up-operator.sh [hunk 9]
```diff
@@ -0,0 +1,127 @@
...
+    NAMESPACE             Kubernetes namespace in which we run our chaos-mesh.
+    DOCKER_REGISTRY       image docker registry
+    IMAGE_TAG             image tag
+    SKIP_IMAGE_BUILD      skip build and push images
+
+EOF
+}
+
+
+installOnly=false
```
Comment (cwen0) [inline hack/local-up-operator.sh:49]: How about renaming it to `dependency_only`?

**Review by cwen0 — COMMENTED** [review]
Rest LGTM

File: install.sh (hunk 7 above)
Comment (mikechengwei) [reply install.sh:618]: We must add `{}` when more than 10 parameters in shell.

File: CONTRIBUTING.md (hunk 6 above)
Comment (mikechengwei) [reply CONTRIBUTING.md:130]: ok

File: hack/local-up-operator.sh (hunk 9 above)
Comment (mikechengwei) [reply hack/local-up-operator.sh:49]: ok

**Comment by mikechengwei** [comment]
@cwen0 @WangXiangUSTC PLTA

**Review by cwen0 — APPROVED** [review]
Good job! LGTM

**Review by zhouqiang-cl — APPROVED** [review]
LGTM

**Pull request merged**